src/sync/sync_session.hpp

```diff
@@ -213,6 +213,13 @@ class SyncSession : public std::enable_shared_from_this<SyncSession> {
     // not make the session reconnect.
     void set_multiplex_identifier(std::string multiplex_identity);
 
+    // See SyncConfig::url_prefix
+    //
+    // This method allows to override this value after the session is created but before it is bound
+    // because of Realm Cloud's token refresh service returning the sync worker ingress path with the token response.
+    // Prefer using the SyncConfig field in all other cases.
+    void set_url_prefix(std::string url_prefix);
```

Comment:
Since this value comes with the refresh token, shouldn't it just be an optional parameter to refresh_access_token()?

Reply:
I modeled this after `set_multiplex_identifier` which is similarly called with a value from the access token refresh response (the sync label, in this case).

I don't think it makes sense for the parameter to be passed on every call to `refresh_access_token()` - it's only needed once, when the session is initially bound after receiving the initial access token from the server. I am not sure, however, what's the best point to insert this in, so I just did what `set_multiplex_identifier` does.